src/Http/ServerRequest.php

```diff
@@ -1268,11 +1268,10 @@ public function getData(?string $name = null, mixed $default = null): mixed
         if ($name === null) {
             return $this->data;
         }
-        if (!is_array($this->data) && $name) {
+        if (!is_array($this->data)) {
```

Comment:
We no longer care about `''` here, right? It shouldn't be used for anything meaningful.

Reply:
Don't see a practical case of having the data array with empty string as key. At least not possible through a form submission.

Reply:
Err if data is non-array than the name becomes irrelevant anyway since `Hash::extract()` can't be used anymore.

Reply:
agreed.